src/deadline/job_attachments/README.md

```diff
@@ -33,6 +33,47 @@ RootPrefix/
 
 [ja-security]: https://docs.aws.amazon.com/deadline-cloud/latest/userguide/security-best-practices.html#job-attachment-queues
 
+### S3 Key Format
+
+#### Data Files
+
+Data files in the job attachments system are stored using a content-addressable approach. The S3 key format for data files is:
+
+```
+RootPrefix/Data/<file_hash>.<hash_algorithm>
+```
+
+For example:
+```
+my-deadline-prefix/Data/a1b2c3d4e5f6g7h8i9j0.xxh128
+```
+
+Where:
+- `<file_hash>` is the hash of the file contents
+- `<hash_algorithm>` is the algorithm used to generate the hash (currently "xxh128")
+
+This content-addressable approach ensures that identical files are only stored once, regardless of their original filenames or paths.
+
+#### Manifest Files
+
+Manifests are stored for both job inputs and task outputs. The S3 key for the input manifest for a job can be found by [calling GetJob](https://docs.aws.amazon.com/deadline-cloud/latest/APIReference/API_GetJob.html#API_GetJob_ResponseSyntax) and looking in the response under `attachments.manifests.rootPath`. See the [developer guide on job attachments](https://docs.aws.amazon.com/deadline-cloud/latest/developerguide/run-jobs-job-attachments.html#job-attachments-in-depth) for more info. While the input manifest location is not formally defined, in the current job attachments implementation it follows the pattern:
+```
+RootPrefix/Manifests/<farm_id>/<queue_id>/Inputs/<guid>/<manifest_hash>_input
+```
+
+Output manifests for tasks are stored under:
+```
+RootPrefix/Manifests/<farm_id>/<queue_id>/<job_id>/<step_id>/<task_id>/<timestamp>_<session_action_id>/<manifest_hash>_output
+```
+
+Where:
+- `<farm_id>`, `<queue_id>`, `<job_id>`, `<step_id>`, `<task_id>`, and `<session_action_id>` are the respective identifiers (e.g., farm-1234567890abcdefg)
+- `<guid>` is a randomly generated identifier for the input manifest
+- `<manifest_hash>` is a hash derived from the source root path
+- `<timestamp>` is an ISO8601 timestamp
+
+Each manifest file contains metadata in its S3 object properties, including the "asset-root" that specifies the local root path where files should be placed when downloaded.
```

Comment:
Maybe mention "asset-root-json" as well. There are characters permitted in paths that are not allowed metadata values.

Reply:
Ah, I didn't realize that was the motivation for "asset-root-json". Is "asset-root-json" always preferable to "asset-root"?

Reply:
You can see the logic introduced for it in https://github.com/aws-deadline/deadline-cloud/pull/492/files#diff-170e0b44232d2df6abb70f90365e2e1811b0ddf17d4f447e4c39f35fe99b50c9R530 that best-effort stores it as "asset-root" and then "asset-root-json".